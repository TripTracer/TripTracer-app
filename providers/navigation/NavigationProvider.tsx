import * as React from 'react';

import { createMaterialBottomTabNavigator } from '@react-navigation/material-bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';

import HomeScreen from '../../screens/Home';
import { RootTabParamList } from '../../utils/types';
import SettingsStackScreen from './stacks/SettingsStack';

const Tab = createMaterialBottomTabNavigator<RootTabParamList>();

function NavigationProvider() {
  return (
    <NavigationContainer>
      <Tab.Navigator initialRouteName='Home'>
        <Tab.Screen
          name='Home'
          component={HomeScreen}
          options={{ title: 'Home' }}
        />
        <Tab.Screen
          name='SettingsStack'
          component={SettingsStackScreen}
          options={{ title: 'Settings' }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
}

export default NavigationProvider;