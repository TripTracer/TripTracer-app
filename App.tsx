import 'react-native-gesture-handler';

import { StatusBar } from 'expo-status-bar';

import I18next from './providers/i18next/I18Next';
import NavigationProvider from './providers/navigation/NavigationProvider';
import QueryProvider from './providers/reactQuery/ReactQuery';
import ThemeProvider from './providers/theme/Theme';

export default function App() {
  return (
    <I18next>
      <QueryProvider>
        <ThemeProvider>
          <NavigationProvider />
          <StatusBar style='auto' />
        </ThemeProvider>
      </QueryProvider>
    </I18next>
  );
}
