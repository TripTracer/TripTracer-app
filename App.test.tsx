import React from 'react';
<<<<<<< HEAD
=======
import renderer from 'react-test-renderer';
<<<<<<< HEAD
>>>>>>> 07c3b61 (feat: theme unit tests added)
=======
>>>>>>> f46e10f (feat: theme unit tests added)
import App from './App';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react-native';
import QueryProvider from './providers/reactQuery/ReactQuery';
import ThemeProvider from './providers/theme/Theme';
import NavigationProvider from './providers/navigation/NavigationProvider';
import { StatusBar } from 'expo-status-bar';
import '@jest/globals'; // Add this line

jest.mock('./providers/reactQuery/ReactQuery', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('./providers/theme/Theme', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('./providers/navigation/NavigationProvider', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('expo-status-bar', () => ({
  __esModule: true,
  StatusBar: jest.fn(),
}));
describe('<App />', () => {
<<<<<<< HEAD
<<<<<<< HEAD
  it('should render the app with necessary components', () => {
    render(<App />);
    waitFor(() => {
      expect(QueryProvider).toHaveBeenCalled();
      expect(ThemeProvider).toHaveBeenCalled();
      expect(NavigationProvider).toHaveBeenCalled();
      expect(StatusBar).toHaveBeenCalled();
    });
  });
  it('should render nothing if font loading fails', () => {
    render(<App />);
    waitFor(() => {
      expect(App).not.toHaveBeenCalled();
    });
=======
  it('should render the app with the necessary components', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not render the QueryProvider, ThemeProvider, and NavigationProvider components', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
>>>>>>> 07c3b61 (feat: theme unit tests added)
=======
  it('should render the app with the necessary components', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not render the QueryProvider, ThemeProvider, and NavigationProvider components', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
>>>>>>> f46e10f (feat: theme unit tests added)
  });
});
