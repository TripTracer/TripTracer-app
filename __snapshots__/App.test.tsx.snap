// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should not render the QueryProvider, ThemeProvider, and NavigationProvider components 1`] = `null`;

exports[`<App /> should render the app with the necessary components 1`] = `null`;
